HEN_HOUSE/src/EEMF_macros.mortran

```diff
@@ -0,0 +1,2396 @@
+	prllIntSum, perpIntSum, xPrllInt, xPerpInt, momTot_EF, momPerp_EF,
+	momPrll_EF, tperp0_em, VEF_initial, VEF_final, StuckLoopCounter, xSave,
+	uem1,vem1,wem1,rem1,uem2,vem2,wem2,rem2,uemf1,vemf1,wemf1,uem3,vem3,wem3,
+	uem4,vem4,wem4;
+
+	$INTEGER irnew_em, palg, show_B;
+
+	$LOGICAL  vstep_changed, thirdOrder, deltaTCalculated, EMtransportActive,
+	EMVacuumDone, EFieldTransport, BFieldTransport;
+
+;};
+
+
+"*****************************************************************************"
+"*****************************************************************************"
+
+
+"*****************************************************************************"
+" ********     	Electric and magnetic field definitions 	************* "
+"*****************************************************************************"
+"!!!NOTE!!! -- needs to be defined by the user if input file defn is missing"
+REPLACE{$GET_EM_FIELD(#, #, #);}WITH{;
+"Call to get the value of the B and E field at the x,y,z position"
+"This macro definition uses the EGSnrc EM field inputs from the input file"
+"If this is missing the user's installation, the BxIN etc. variables should be"
+"replaced or a new field should be defined"
+
+  "Magnetic field:"
+  Bx = BxIN;By = ByIN;Bz = BzIN;
+  "Electric field:"
+  Ex = ExIN;Ey = EyIN;Ez = EzIN;
+
+  "delta_u definition from the user"
+  delta_u = EMLMTIN;
+}
+
+REPLACE{$setEFPotential(#, #, #, #);}WITH{
+" EXAMPLE Electric field potential definition"
+" The potential here is a sample potential, and should be defined by the user"
+	r0_em =sqrt({P2}*{P2} + ({P3} + 1.5)**2);
+IF(r0_em>0.35)[;
+	"calculate cos_theta:"
+	By = ({P3} + 1.5)/r0_em;
+"calculate the potential at this point (the 1000 is a conversion to Volts)"
+"E0=1100kV/cm"
+"Final units of {P4} should be volts"
+	{P4} = -1000*1100.*.35*log(r0_em/.35)*(1-.2272727*By-.4090909*By**2);
+	By=0.;
+]ELSE[
+	"potential is zero at the surface!"
+	{P4} = 0.;
+]
+}
+"******************************************************************************"
+"******************************************************************************"
+
+
+"******************************************************************************"
+	    "*************** STEP SIZE RESTRICTION ***************"
+"******************************************************************************"
+"This MACROS allows the control of the accuracy of the calculations via step  "
+"size shortening restrictions. These restrictions are based on an approximate "
+"first order EM field influence, and attempt to control for the angular devia-"
+"-tion, the energy change, and the change in the magnitude of the fields      "
+
+REPLACE{$EMFIELD_INITIATE_SET_TUSTEP;}WITH{;
+  "Set the magnetic and electric field values to a default of 0:"
+  "Magnetic field:"
+  Bx = 0.;By = 0.;Bz = 0.;
+  B_amp = 0.;
+
+  "set the default delta_u"
+  delta_u = $deltaUDefault;
+
+  "Electric field:"
+  "Convert V/cm to V/m (the code uses V/m)"
```

Comment:
Seems this comment is outdated: here the electric field is initialized to 0; the adjustment to V/m occurs on lines 284-285.

Reply:
Fixed it.